$OpenBSD: patch-widget_nsShmImage_cpp,v 1.3 2018/09/05 16:19:01 landry Exp $

shmget sometimes still gets called:
https://bugzilla.mozilla.org/show_bug.cgi?id=1376910 ?
https://bugzilla.mozilla.org/show_bug.cgi?id=1438401 ?
also at startup when layers.acceleration.force-enable defaults to false

https://hg.mozilla.org/mozilla-central/rev/3ad8c54c6dc8

Index: widget/nsShmImage.cpp
--- widget/nsShmImage.cpp.orig
+++ widget/nsShmImage.cpp
@@ -67,6 +67,12 @@ nsShmImage::CreateShmSegment()
 {
   size_t size = SharedMemory::PageAlignedSize(mStride * mSize.height);
 
+#if defined(__OpenBSD__) && defined(MOZ_SANDBOX)
+  static mozilla::LazyLogModule sPledgeLog("SandboxPledge");
+  MOZ_LOG(sPledgeLog, mozilla::LogLevel::Debug,
+         ("%s called when pledged\n", __func__));
+  return false;
+#endif
   mShmId = shmget(IPC_PRIVATE, size, IPC_CREAT | 0600);
   if (mShmId == -1) {
     return false;
